File: backend/src/main/java/tom/projects/monkeylog/model/workout/ExerciseRowField.java
package tom.projects.monkeylog.model.workout;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.UuidGenerator;
import tom.projects.monkeylog.model.exercise.ExerciseType;
import tom.projects.monkeylog.model.user.UserOwned;

import java.util.UUID;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class ExerciseRowField implements UserOwned {
    @Id
    @GeneratedValue
    @UuidGenerator
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "exercise_row_id", nullable = false)
    private ExerciseRow exerciseRow;

    private Double value;

    @Enumerated(EnumType.STRING)
    private ExerciseType exerciseType;

    public static ExerciseRowField clone(ExerciseRowField exerciseRowField) {
        ExerciseRowField newExerciseRowField = new ExerciseRowField();
        newExerciseRowField.setExerciseType(exerciseRowField.getExerciseType());
        newExerciseRowField.setValue(exerciseRowField.getValue());

        return newExerciseRowField;
    }

    @Override
    public UUID getUserId() {
        return exerciseRow.getUserId();
    }
}

File: backend/bin/main/tom/com/monkeylog/model/workout/ExerciseRowField.kt
package tom.com.monkeylog.model.workout

import jakarta.persistence.*
import org.hibernate.annotations.UuidGenerator
import tom.com.monkeylog.model.exercise.ExerciseType
import tom.com.monkeylog.model.user.UserOwned
import java.util.*

@Entity
class ExerciseRowField(
    @Id
    @GeneratedValue
    @UuidGenerator
    var id: UUID? = null,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "exercise_row_id", nullable = false)
    var exerciseRow: ExerciseRow,
    var value: Double,
    @Enumerated(EnumType.STRING)
    var exerciseType: ExerciseType,
    override var userId: UUID? = null,
) : UserOwned {

    fun clone(exerciseRow: ExerciseRow): ExerciseRowField {
        return ExerciseRowField(
            exerciseRow = exerciseRow,
            value = value,
            exerciseType = exerciseType,
        )
    }
}
